### PR TITLE
refactor(server): telemetry

### PR DESCRIPTION
**Pull request by jrasm91**

### CODE REVIEW

**Review by danieldietzler — COMMENTED** [review]
Looks really good so far, I quite like it!

**Review by danieldietzler — APPROVED** [review]

File: server/src/repositories/telemetry.repository.ts [hunk 1]
```diff
@@ -33,20 +48,120 @@ class MetricGroupRepository implements IMetricGroupRepository {
...
+        prefix: 'immich',
+        description: `The elapsed time in ${unit} for the ${startCase(className)} to ${propertyName.toLowerCase()}`,
+        unit,
+        valueType: contextBase.ValueType.DOUBLE,
+      });
+
+      descriptor.value = function (...args: any[]) {
+        const start = performance.now();
+        // TODO: wrap with try/catch
+        const result = method.apply(this, args);
```
Comment (mertalev) [inline server/src/repositories/telemetry.repository.ts:152]: Why would you wrap it?

File: server/src/repositories/telemetry.repository.ts [hunk 2]
```diff
@@ -33,20 +48,120 @@ class MetricGroupRepository implements IMetricGroupRepository {
...
 
     this.api = new MetricGroupRepository(metricService).configure({ enabled: apiMetrics });
     this.host = new MetricGroupRepository(metricService).configure({ enabled: hostMetrics });
     this.jobs = new MetricGroupRepository(metricService).configure({ enabled: jobMetrics });
     this.repo = new MetricGroupRepository(metricService).configure({ enabled: repoMetrics });
   }
+
+  setup({ repositories }: { repositories: ClassConstructor<unknown>[] }) {
+    const { telemetry } = this.configRepository.getEnv();
+    if (!telemetry.enabled || !telemetry.repoMetrics) {
```
Comment (mertalev) [inline server/src/repositories/telemetry.repository.ts:112]: This is a slight difference in behavior, right? `IMMICH_METRICS` takes priority over `IMMICH_IO_METRICS` with this change. You mentioned changing this to include/exclude so it doesn't matter much, but if you don't think you'll get to that before the next release then you'd want to update the docs on the new behavior.

**Review by mertalev — APPROVED** [review]

File: server/src/repositories/telemetry.repository.ts (hunk 1 above)
Comment (jrasm91) [reply server/src/repositories/telemetry.repository.ts:152]: Oh you are right. For some reason I thought we did something on error that was missing.

File: server/src/repositories/telemetry.repository.ts (hunk 2 above)
Comment (jrasm91) [reply server/src/repositories/telemetry.repository.ts:112]: Enabled gets set to true here if any are enabled., which is how you had it before. I agree it is confusing though.

Comment (mertalev) [reply server/src/repositories/telemetry.repository.ts:112]: Oh true, I forgot about that too lol. That's fine then

**Pull request merged**